pkg/functionconfig/types.go

```diff
@@ -323,38 +323,39 @@ type Build struct {
-	SecurityContext         *v1.PodSecurityContext  `json:"securityContext,omitempty"`
-	ServiceAccount          string                  `json:"serviceAccount,omitempty"`
-	ScaleToZero             *ScaleToZeroSpec        `json:"scaleToZero,omitempty"`
+	Description               string                  `json:"description,omitempty"`
+	Disable                   bool                    `json:"disable,omitempty"`
+	Publish                   bool                    `json:"publish,omitempty"`
+	Handler                   string                  `json:"handler,omitempty"`
+	Runtime                   string                  `json:"runtime,omitempty"`
+	Env                       []v1.EnvVar             `json:"env,omitempty"`
+	Resources                 v1.ResourceRequirements `json:"resources,omitempty"`
+	Image                     string                  `json:"image,omitempty"`
+	ImageHash                 string                  `json:"imageHash,omitempty"`
+	Replicas                  *int                    `json:"replicas,omitempty"`
+	MinReplicas               *int                    `json:"minReplicas,omitempty"`
+	MaxReplicas               *int                    `json:"maxReplicas,omitempty"`
+	TargetCPU                 int                     `json:"targetCPU,omitempty"`
+	DataBindings              map[string]DataBinding  `json:"dataBindings,omitempty"`
+	Triggers                  map[string]Trigger      `json:"triggers,omitempty"`
+	Volumes                   []Volume                `json:"volumes,omitempty"`
+	Version                   int                     `json:"version,omitempty"`
+	Alias                     string                  `json:"alias,omitempty"`
+	Build                     Build                   `json:"build,omitempty"`
+	RunRegistry               string                  `json:"runRegistry,omitempty"`
+	ImagePullSecrets          string                  `json:"imagePullSecrets,omitempty"`
+	RuntimeAttributes         map[string]interface{}  `json:"runtimeAttributes,omitempty"`
+	LoggerSinks               []LoggerSink            `json:"loggerSinks,omitempty"`
+	DealerURI                 string                  `json:"dealerURI,omitempty"`
+	Platform                  Platform                `json:"platform,omitempty"`
+	ReadinessTimeoutSeconds   int                     `json:"readinessTimeoutSeconds,omitempty"`
+	Avatar                    string                  `json:"avatar,omitempty"`
+	ServiceType               v1.ServiceType          `json:"serviceType,omitempty"`
+	ImagePullPolicy           v1.PullPolicy           `json:"imagePullPolicy,omitempty"`
+	SecurityContext           *v1.PodSecurityContext  `json:"securityContext,omitempty"`
+	ServiceAccount            string                  `json:"serviceAccount,omitempty"`
+	ScaleToZero               *ScaleToZeroSpec        `json:"scaleToZero,omitempty"`
+	DisableDefaultHttpTrigger *bool                   `json:"disableDefaultHttpTrigger,omitempty"`
```

Comment:
Add a comment that when this is nil, the value is taken from the platform config, and when set explicitly it has priority